SWLOR.Game.Server/Feature/AbilityDefinition/Beastmaster/RewardAbilityDefinition.cs

```diff
@@ -77,9 +77,10 @@ private string Validation(uint activator)
         private void Impact(uint activator, int baseHealingAmount)
         {
             var willBonus = GetAbilityModifier(AbilityType.Social, activator);
-            var amount = baseHealingAmount + willBonus * 8 + Random.D10(1);
-
             var beast = GetAssociate(AssociateType.Henchman, activator);
+            var maxhp = GetMaxHitPoints(beast);
+            var amount = baseHealingAmount + willBonus * 10 + (maxhp / 5);
```

Comment:
Add a Random() call to the end of this just so it isn't a static number every single time.